feat: add `ndarray/some-by`

headlessNode opened the pull request:
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: passed
  - task: lint_package_json status: passed
  - task: lint_repl_help status: passed
  - task: lint_javascript_src status: passed
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: passed
  - task: lint_javascript_tests status: passed
  - task: lint_javascript_benchmarks status: passed
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: na
  - task: lint_c_examples status: na
  - task: lint_c_benchmarks status: na
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: passed
  - task: lint_typescript_tests status: passed
  - task: lint_license_headers status: passed ---

Progresses #2656 .

## Description

> What is the purpose of this pull request?

This pull request:

-   add `ndarray/some-by`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   progresses #2656

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

headlessNode committed:
feat: add ndarray/some-by

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

headlessNode committed:
fix: bugs & docs

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

headlessNode commented on lib/node_modules/@stdlib/ndarray/some-by/lib/main.js:
```diff
@@ -0,0 +1,231 @@
+var validate = require( './validate.js' );
+
+
+// VARIABLES //
+
+var TYPES = [
+	'int8',
+	'int16',
+	'int32',
+	'uint8',
+	'uint8c',
+	'uint16',
+	'uint32'
+];
+
```
@kgryte Is this okay for allowed types for the `n` if its an ndarray?

headlessNode replied:
Casting didn't make sense to me for `some-by`

kgryte replied:
Do not hard code a list of data types. Use `ndarray/dtypes` and provide a "kind" argument.

kgryte replied:
...otherwise, you are just imposing a future maintenance burden.

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/README.md:
```diff
@@ -0,0 +1,389 @@
+
+# someBy
+
+> Test whether at least `n` elements in an [`ndarray`][@stdlib/ndarray/ctor] pass a test implemented by a predicate function along one or more dimensions.
```
```suggestion
> Test whether at least `n` elements along one or more [`ndarray`][@stdlib/ndarray/ctor] dimensions pass a test implemented by a predicate function.
```

Applies here and everywhere.

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/README.md:
```diff
@@ -0,0 +1,389 @@
+var ox = 1;
+
+// Create an input ndarray:
+var x = new ndarray( 'float64', xbuf, sh, sx, ox, 'row-major' );
```
As in `ndarray/count-if`, use `ndarray/array` in order to simplify ndarray creation in public docs. As this is a non-base package, we should aim to reduce visual noise in examples, as these APIs are more user facing. That means providing a nested array so that the shape can be inferred, etc.

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/README.md:
```diff
@@ -0,0 +1,389 @@
+The function accepts the following arguments:
+
+-   **x**: input [`ndarray`][@stdlib/ndarray/ctor].
+-   **n**: number of elements which must pass the test implemented by a predicate function. May be either a scalar or an [`ndarray`][@stdlib/ndarray/ctor].
```
As in `blas/ext/cusum`, you need to explicitly state that `n` must be broadcast-compatible with `x`. Furthermore, you need to state the allowed data types. Again, please think about this from the user's POV. If you don't say what the allowed dtypes are, how are they going to know?

```suggestion
-   **n**: number of elements which must pass the test implemented by a predicate function. May be either a scalar or an [`ndarray`][@stdlib/ndarray/ctor]. Must be [broadcast-compatible][@stdlib/ndarray/base/broadcast-shapes] with the non-reduced dimensions of input [`ndarray`][@stdlib/ndarray/ctor]. Must have an integer [data type][@stdlib/ndarray/dtypes].
```

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/README.md:
```diff
@@ -0,0 +1,389 @@
+    return value > 0.0;
+}
+
+// Create a data buffer:
```
Refactor this example.

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/README.md:
```diff
@@ -0,0 +1,389 @@
+
+By default, the function returns an [`ndarray`][@stdlib/ndarray/ctor] having a shape matching only the non-reduced dimensions of the input [`ndarray`][@stdlib/ndarray/ctor] (i.e., the reduced dimensions are dropped). To include the reduced dimensions as singleton dimensions in the output [`ndarray`][@stdlib/ndarray/ctor], set the `keepdims` option to `true`.
+
+<!-- eslint-disable max-len -->
```
When you refactor the examples, you should be able to remove this and similar comments.

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/README.md:
```diff
@@ -0,0 +1,389 @@
+    return value > 0.0;
+}
+
+// Create a data buffer:
```
Update example.

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/README.md:
```diff
@@ -0,0 +1,389 @@
+
+To set the predicate function execution context, provide a `thisArg`.
+
+<!-- eslint-disable no-invalid-this, max-len -->
```
```suggestion
<!-- eslint-disable no-invalid-this -->
```

headlessNode replied:
here I added this because I was having a `length` lint error when pushing

kgryte replied:
Yes, but this will go away once you refactor the examples. This is a reminder to update this comment.

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/README.md:
```diff
@@ -0,0 +1,389 @@
+
+#### someBy.assign( x, n, out\[, options], predicate\[, thisArg] )
+
+Test whether at least `n` elements in an [`ndarray`][@stdlib/ndarray/ctor] pass a test implemented by a predicate function along one or more dimensions and assigns results to a provided output [`ndarray`][@stdlib/ndarray/ctor].
```
```suggestion
Tests whether at least `n` elements in an [`ndarray`][@stdlib/ndarray/ctor] pass a test implemented by a predicate function along one or more dimensions and assigns results to a provided output [`ndarray`][@stdlib/ndarray/ctor].
```

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/README.md:
```diff
@@ -0,0 +1,389 @@
+The function accepts the following arguments:
+
+-   **x**: input [`ndarray`][@stdlib/ndarray/ctor].
+-   **n**: number of elements which must pass the test implemented by a predicate function. May be either a scalar or an [`ndarray`][@stdlib/ndarray/ctor].
```
```suggestion
-   **n**: number of elements which must pass the test implemented by a predicate function. May be either a scalar or an [`ndarray`][@stdlib/ndarray/ctor].  Must be [broadcast-compatible][@stdlib/ndarray/base/broadcast-shapes] with the non-reduced dimensions of input [`ndarray`][@stdlib/ndarray/ctor]. Must have an integer [data type][@stdlib/ndarray/dtypes].
```

kgryte replied:
You'll need to add the link to `ndarray/base/broadcast-shapes`.

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/README.md:
```diff
@@ -0,0 +1,389 @@
+    return value > 0.0;
+}
+
+// Create a data buffer:
```
Refactor example.

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/docs/repl.txt:
```diff
@@ -0,0 +1,121 @@
+    x: ndarray
+        Input ndarray.
+
+    n: ndarray|any
```
```suggestion
    n: ndarray|integer
```

Why does `any` make sense here?

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/docs/repl.txt:
```diff
@@ -0,0 +1,121 @@
+        Input ndarray.
+
+    n: ndarray|any
+        Number of elements which must pass the test implemented by a predicate
```
You need to mention broadcast-compatibility and allowed dtypes.

kgryte replied:
You didn't update this description to indicate allowed dtypes.

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/docs/repl.txt:
```diff
@@ -0,0 +1,121 @@
+
+    Examples
+    --------
+    > var xbuf = new {{alias:@stdlib/array/float64}}( [ 1.0, 2.0, 3.0, 4.0 ] );
```
See `ndarray/count-if` for simplified array creation.

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/docs/repl.txt:
```diff
@@ -0,0 +1,121 @@
+
+    n: ndarray|any
+        Number of elements which must pass the test implemented by a predicate
+        function.
```
Same comments.

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/docs/repl.txt:
```diff
@@ -0,0 +1,121 @@
+
+    Examples
+    --------
+    > var xbuf = new {{alias:@stdlib/array/float64}}( [ 1.0, 2.0, 3.0, 4.0 ] );
```
Same comments.

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/docs/types/index.d.ts:
```diff
@@ -0,0 +1,344 @@
+	* var v = out.get();
+	* // returns true
+	*/
+	<T = unknown, U = unknown>( x: ndarray, n: ndarray | unknown, predicate: Predicate<T,U>, thisArg?: ThisParameterType<Predicate<T, U>> ): boolndarray;
```
We can do better here.

```suggestion
	<T = unknown, U = unknown>( x: ndarray, n: integerndarray | number, predicate: Predicate<T,U>, thisArg?: ThisParameterType<Predicate<T, U>> ): boolndarray;
```

Applies here and below.

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/docs/types/test.ts:
```diff
@@ -0,0 +1,387 @@
+	someBy( [ 1 ], 2, {}, clbk, {} ); // $ExpectError
+	someBy( ( x: number ): number => x, 2, {}, clbk, {} ); // $ExpectError
+}
+
```
You are missing tests for the second argument.

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/docs/types/test.ts:
```diff
@@ -0,0 +1,387 @@
+	someBy.assign( [ 1 ], 2, y, {}, clbk, {} ); // $ExpectError
+	someBy.assign( ( x: number ): number => x, 2, y, {}, clbk, {} ); // $ExpectError
+}
+
```
You are missing tests for the second argument.

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/lib/assign.js:
```diff
@@ -0,0 +1,205 @@
+
+// VARIABLES //
+
+var TYPES = [
```
No hardcoding dtypes. This just increases maintenance burden if and when we add `u/int64` support.

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/lib/assign.js:
```diff
@@ -0,0 +1,205 @@
+* Tests whether at least `n` elements in an ndarray pass a test implemented by a predicate function along one or more dimensions.
+*
+* @param {ndarray} x - input ndarray
+* @param {(ndarray|*)} n - number of elements which must pass the test
```
```suggestion
* @param {(ndarray|integer)} n - number of elements which must pass the test
```

Why does `*` make sense?

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/lib/assign.js:
```diff
@@ -0,0 +1,205 @@
+	N = ndims( x );
+	nargs = arguments.length;
+
+	if ( nargs === 4 ) {
```
```suggestion
	if ( nargs < 5 ) {
```

For explicitly checking for exact equality, you don't cover all cases. What happens if a user mistakenly provides only 3 arguments?

kgryte replied:
Also, as in `ndarray/count-if`, add comments regarding the signatures. This helps you (a) reason about the code and (b) helps future readers.

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/lib/assign.js:
```diff
@@ -0,0 +1,205 @@
+				opts.dims = zeroTo( N );
+			}
+		}
+	} else if ( nargs === 6 ) {
```
```suggestion
	} else if ( nargs > 5 ) {
```

Same thing.

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/lib/assign.js:
```diff
@@ -0,0 +1,205 @@
+		cb = options;
+		ctx = thisArg;
+		opts = objectAssign( {}, defaults );
+		opts.dims = zeroTo( N );
```
Don't repeat things like this. See `ndarray/count-if`. All of this logic here and below needs to be cleaned up.

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/lib/assign.js:
```diff
@@ -0,0 +1,205 @@
+			}
+			cb = predicate;
+			ctx = thisArg;
+			opts = objectAssign( {}, defaults, options );
+			err = validate( opts, N, options );
+			if ( err ) {
+				throw err;
+			}
+			if ( opts.dims === null ) {
+				opts.dims = zeroTo( N );
+			}
```
Why does this need to be repeated here and again at L168. Again, think about simplifying code structure.

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/lib/assign.js:
```diff
@@ -0,0 +1,205 @@
+	ord = getOrder( x );
+
+	if ( isndarrayLike( n ) ) {
+		if ( !contains( TYPES, getDtype( n ) ) ) {
```
You don't need to use contains. Use `ndarray/base/assert/is-integer-data-type` (or whatever the pkg name is).

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/lib/assign.js:
```diff
@@ -0,0 +1,205 @@
+
+	if ( isndarrayLike( n ) ) {
+		if ( !contains( TYPES, getDtype( n ) ) ) {
+			throw new TypeError( format( 'invalid argument. Second argument must be an ndarray-like object having an integer data type. Value: `%s`.', n ) );
```
```suggestion
			throw new TypeError( format( 'invalid argument. Second argument must have an integer data type. Value: `%s`.', n ) );
```

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/lib/assign.js:
```diff
@@ -0,0 +1,205 @@
+		if ( !isInteger( n ) ) {
+			throw new TypeError( format( 'invalid argument. Second argument must be an integer or an ndarray-like object. Value: `%s`.', n ) );
+		}
+		v = broadcastScalar( n, 'generic', getShape( y ), ord );
```
Do not use `generic` here. If you use `generic`, you preclude us from being able to call into C into the future, as "generic" arrays cannot be easily supported in native code. Use the default `integer_index` dtype. See `ndarray/count-if` for how to resolve that dtype.

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/lib/main.js:
```diff
@@ -0,0 +1,231 @@
+* Tests whether at least `n` elements in an ndarray pass a test implemented by a predicate function along one or more dimensions.
+*
+* @param {ndarray} x - input ndarray
+* @param {(ndarray|*)} n - number of elements which must pass a test
```
Same comments.

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/lib/main.js:
```diff
@@ -0,0 +1,231 @@
+* @throws {TypeError} first argument must be an ndarray-like object
+* @throws {TypeError} second argument must be an ndarray-like object or a scalar value
+* @throws {TypeError} options argument must be an object
+* @throws {TypeError} predicate must be a function
```
```suggestion
* @throws {TypeError} callback argument must be a function
```

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/lib/assign.js:
```diff
@@ -0,0 +1,205 @@
+* @throws {TypeError} third argument must be an ndarray-like object
+* @throws {TypeError} options argument must be an object
+* @throws {Error} must provide valid options
+* @throws {TypeError} predicate must be a function
```
```suggestion
* @throws {TypeError} callback argument must be a function
```

Use similar description as in `count-if`.

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/lib/main.js:
```diff
@@ -0,0 +1,231 @@
+	N = shx.length;
+	nargs = arguments.length;
+
+	if ( nargs === 3 ) {
```
All the same comments regarding argument juggling.

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/lib/main.js:
```diff
@@ -0,0 +1,231 @@
+
+	if ( nargs === 3 ) {
+		if ( !isFunction( options ) ) {
+			throw new TypeError( format( 'invalid argument. Predicate must be a function. Value: `%s`.', options ) );
```
See `ndarray/count-if`. Applies to similar error messages below.

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/lib/main.js:
```diff
@@ -0,0 +1,231 @@
+	ord = getOrder( x );
+
+	if ( isndarrayLike( n ) ) {
+		if ( !contains( TYPES, getDtype( n ) ) ) {
```
Same comments.

kgryte commented on lib/node_modules/@stdlib/ndarray/some-by/lib/main.js:
```diff
@@ -0,0 +1,231 @@
+		if ( !isInteger( n ) ) {
+			throw new TypeError( format( 'invalid argument. Second argument must be an integer or an ndarray-like object. Value: `%s`.', n ) );
+		}
+		v = broadcastScalar( n, 'generic', shy, ord );
```
Same comments.

headlessNode committed:
fix: apply suggestions from code review

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

headlessNode commented on lib/node_modules/@stdlib/ndarray/some-by/README.md:
```diff
@@ -0,0 +1,315 @@
+
+Tests whether at least `n` elements along one or more [`ndarray`][@stdlib/ndarray/ctor] dimensions pass a test implemented by a predicate function.
+
+<!-- eslint-disable max-len -->
```
```suggestion
```

headlessNode commented on lib/node_modules/@stdlib/ndarray/some-by/README.md:
```diff
@@ -0,0 +1,315 @@
+
+Tests whether at least `n` elements along one or more [`ndarray`][@stdlib/ndarray/ctor] dimensions pass a test implemented by a predicate function and assigns results to a provided output [`ndarray`][@stdlib/ndarray/ctor].
+
+<!-- eslint-disable max-len -->
```
```suggestion
```

headlessNode commented on lib/node_modules/@stdlib/ndarray/some-by/docs/repl.txt:
```diff
@@ -0,0 +1,113 @@
+
+    n: ndarray|integer
+        Number of elements which must pass the test implemented by a predicate
+        function. Must be brodcast compatible with the non-reduced dimensions of
```
```suggestion
        function. Must be broadcast compatible with the non-reduced dimensions of
```

headlessNode commented on lib/node_modules/@stdlib/ndarray/some-by/docs/repl.txt:
```diff
@@ -0,0 +1,113 @@
+
+    n: ndarray|integer
+        Number of elements which must pass the test implemented by a predicate
+        function. Must be brodcast compatible with the non-reduced dimensions of
```
```suggestion
        function. Must be broadcast compatible with the non-reduced dimensions of
```

headlessNode committed:
Apply suggestions from code review

Signed-off-by: Muhammad Haris <101793258+headlessNode@users.noreply.github.com>

headlessNode committed:
docs: fix line length

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

headlessNode commented:
@kgryte ready 🤞🏻

kgryte committed:
chore: clean-up

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

kgryte committed:
test: update dtype

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

kgryte reviewed (approved):
LGTM after clean-up, especially wrt argument juggling.

Planeshifter commented on lib/node_modules/@stdlib/ndarray/some-by/lib/assign.js:
```diff
@@ -0,0 +1,190 @@
+	ord = getOrder( x );
+
+	if ( isndarrayLike( n ) ) {
+		if ( isIntegerDataType( getDtype( n ) ) ) {
```
@kgryte @headlessNode Should the condition be inverted here?

kgryte replied:
Yes. Thank you for catching.
